pallets/parachain-staking/src/lib.rs

```diff
@@ -1444,6 +1443,68 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		/// Notify a collator is inactive during MaxOfflineRounds
+		#[pallet::call_index(29)]
+		#[pallet::weight(
+			T::DbWeight::get().reads_writes(0 as u64, 0 as u64)
+		)]
+		pub fn notify_inactive_collator(
+			origin: OriginFor<T>,
+			collator: T::AccountId,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let collators = <SelectedCandidates<T>>::get();
+			let max_collators = <TotalSelected<T>>::get();
+
+			// Check collators length is not below or eq to 66% of max_collators.
+			// We use saturating logic here with (2/3)
+			// as it is dangerous to use floating point numbers directly.
+			ensure!(
+				collators.len() * 3 > (max_collators * 2) as usize,
+				<Error<T>>::TooLowCollatorCountToNotifyAsInactive
+			);
+
+			let round_info = <Round<T>>::get();
+			let max_offline_rounds = T::MaxOfflineRounds::get();
+
+			// Take last round to have rounds_to_check = [7,8,9]
+			// in case we are in round 10 for instance
+			let round = round_info.current.saturating_sub(1);
+
+			let mut rounds_to_check: Vec<RoundIndex> = vec![];
+			for round_index in (0..max_offline_rounds).rev() {
+				rounds_to_check.push(round.saturating_sub(round_index));
+			}
+
+			// If this counter is eq to max_offline_rounds,
+			// the collator should be notified as inactive
+			let mut inactive_counter: RoundIndex = RoundIndex::default();
```

Comment:
RoundIndex is known here to be type `u32`, I would initialize it to `0u32` instead of `Default`.